### PR TITLE
Beining/fix done command bugs

### LINKED ISSUE #220: [PE-D] Missing error message for 'done' command
Just a small nit to pick, but when a very large integer (larger than 2^31 - 1) is used to indicate done minutes, there is no error message and the command is simply ignored.

Steps to reproduce: 

Call `done 1:30000000000000000000000`, assuming the task at index 1 is a deadline.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: sogggy/ped#6

### LINKED ISSUE #236: [PE-D] Invalid duration for time taken to be complete can be added.
Steps to reproduce:
1) Add a deadline task as input, for example:
deadline title: Tutorial 1 desc: finish up tutorial 1 for CS1234 datetime: 01-01-0001 10:00 tag: CS1234

2) Mark it as done with the following input:
done 1:0:-05 (as seen below)

Expected:
Time taken should be alphanumeric, and therefore there should be an error marking as done.

Actual:

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: yongmingyang/ped#5

### PR DESCRIPTION
**Pull request by dearvae**
fix #236 fix #220

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/DoneCommandParser.java [hunk 1]
```diff
@@ -30,11 +31,16 @@ public DoneCommand parse(String args) throws ParseException {
...
+                if (!Duration.isValidDuration(durations[i])) {
+                    throw new ParseException(Duration.INVALID_DURATION_FORMAT);
+                }
             }
             return new DoneCommand(indexes, durations);
         } catch (ParseException pe) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, pe.getMessage(), DoneCommand.MESSAGE_USAGE), pe);
+        } catch (NumberFormatException ne) {
+            throw new ParseException("should be a positive number and the maximum duration is 2147483647 minutes");
```
Comment (MarcTzh) [inline src/main/java/seedu/address/logic/parser/DoneCommandParser.java]: Magic number

Comment (MarcTzh) [reply src/main/java/seedu/address/logic/parser/DoneCommandParser.java]: MAX_INT = .....
then string format

File: src/test/java/seedu/address/model/task/deadline/DurationTest.java [hunk 2]
```diff
@@ -19,9 +19,10 @@ public void isValidTitle() {
```
Comment (MarcTzh) [inline src/test/java/seedu/address/model/task/deadline/DurationTest.java:19]: Why is this called isValidTitle??? Rename

File: src/test/java/seedu/address/model/task/deadline/DurationTest.java [hunk 3]
```diff
@@ -19,9 +19,10 @@ public void isValidTitle() {
 
         // invalid duration
         assertFalse(Duration.isValidDuration(-10)); // negative number
```
Comment (MarcTzh) [inline src/test/java/seedu/address/model/task/deadline/DurationTest.java:21]: Test a few more values in each equivalence partition and especially near boundary values. Please include values such as 0, -1, some valid values and values near MAX_INT

**Review by BobbyZhouZijian — APPROVED** [review]
LGTM
